src/main/java/knox/spring/data/neo4j/controller/KnoxController.java

```diff
@@ -522,6 +522,38 @@ public KnoxController(DesignSpaceService designSpaceService) {
 		}
 	}
 
+	/**
+	 * @api {post} /designSpace/reverse Reverse
+	 * @apiName reverseDesignSpaces
+	 * @apiGroup DesignSpace
+	 *
+	 * @apiParam {String[]} inputSpaceIDs IDs for the input design spaces to be reversed.
+	 * @apiParam {String} [outputSpaceID] ID for the output design space resulting from reverse.  If omitted, then the result is
+	 *  stored in the first input design space.
+	 *
+	 * @apiDescription Reverse the edges from input design spaces.
+	 */
+	@RequestMapping(value = "/designSpace/reverse", method = RequestMethod.POST)
+	public ResponseEntity<String> reverseDesignSpaces(@RequestParam(value = "inputSpaceID", required = true) String inputSpaceID,
+													@RequestParam(value = "outputSpaceID", required = false) String outputSpaceID) {
+		try {
+			long startTime = System.nanoTime();
+
+			if (outputSpaceID == null) {
```

Comment:
I'd prefer to see this if-else handled in the function itself, instead of two separate functions.

Reply:
If the output space is null, in the file `DesignSpaceService.java`, the functions usually choose the first element in the inputBranches as the output space. I think my charges here are consistent?

Reply:
I see that in the `DesignSpaceService.java`, but when you test on the UI, it will throw an error if the outputBranch is empty. But the change you made, where you put get .get(0) in `knox.js`, is inconsistent with the rest of the repo.